### PR TITLE
MAINTAINERS: Added myself as JSON maintainer

### LINKED ISSUE #38477: json: JSON Library Orphaned, Request to Become a Maintainer
Apologies in advance if the format is incorrect, didn't find a template or previous issue for Orphaned code. 

The JSON library appears to be orphaned (technically JSON Web Token, because the JSON library is a subsystem of the OS) according to [MAINTAINERS.YML](https://github.com/zephyrproject-rtos/zephyr/blob/main/MAINTAINERS.yml), @mrfuchs appears a collaborator (@M1cha has also recently submitted a couple of PRs).

The library currently isn't fully compatible with C++. I have a couple of PRs open for improvements #37941 #38052 #38061. I'm looking to improve the general functionality, maintainability and readability of the code. It's not a very easy code to follow and there's some architectural shortcomings in my opinion.

What's the process to become a maintainer for these files ( [/lib/os/json.c](https://github.com/zephyrproject-rtos/zephyr/blob/main/lib/os/json.c) and [include/data/json.h](https://github.com/zephyrproject-rtos/zephyr/blob/main/include/data/json.h))? I know there's a lot of work to be done in the testing side, I don't have much experience with twister and unit test on Zephyr but I'm willing to learn.

Given that this is part of the OS I should also probably aim for MISRA-C compliance according to #9552. 

Thanks & Best Regards,
Ramiro

### PR DESCRIPTION
**Pull request by Sir-Branch**
Added myself as maintainer for the JSON library, added references to the `lib/os/json.c` source file and the `include/data/` folder which contains the json.h and jwt.h headers

Linked to opened issue #38477 

Linked to open PRs  #37941 #38052 #38061

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]
+1 from me

**Comment by cfriedt** [comment]
@mrfuchs - please review the roles and responsibilities of collaborators and maintainers.

https://docs.zephyrproject.org/latest/development_process/project_roles.html

Could you please first add yourself first as a collaborator? Thanks for your willingness to take on maintainership of JSON, because it is quite important. We would be happy to bring up promoting you to maintainer after you can demonstrate a good track record as collaborator and with TSC approval.

**Comment by Sir-Branch** [comment]
@cfriedt Thanks for your help and guidance with this process, sounds good to me. I've updated the commit accordingly.

**Review by henrikbrixandersen — APPROVED** [review]

**Comment by Sir-Branch** [comment]
Hi @carlescufi @cfriedt I know you guys have a ton on your plate and that the JSON library is orphaned so there's not a go to person for reviews. What's the process for reviews in these cases? I have a couple of open PRs and they're starting to block further development.

Thanks for all the help

**Comment by martinjaeger** [comment]
@Sir-Branch I've had a look at some of your PRs. As most of them deal with small refactorings, getting them reviewed might even be faster if the commits were all in one single PR. This has the advantage that you can keep working on that branch yourself and rebase to main once everything got merged.

**Comment by Sir-Branch** [comment]
Appreciate the insight @martinjaeger. I know it sometimes boils down to preference of the reviewer, and given that you've been reviewing lot's of them I'm happy to be having the conversation. I'll try to be more comprehensive in my PRs going forward.

**Pull request merged**
